### PR TITLE
Add dataset.delete_scene

### PR DESCRIPTION
**Pull request by gatli**
Adds Scene deletion through the SDK.

Still waiting for the endpoint to be deployed!

### CODE REVIEW

File: nucleus/dataset.py [hunk 1]
```diff
@@ -738,6 +740,11 @@ def delete_item(self, reference_id: str) -> dict:
             requests.delete,
         )
 
+    @sanitize_string_args
+    def delete_scene(self, reference_id: str):
```
Comment (pfmark) [inline nucleus/dataset.py:744]: Wondering if it makes more sense to just have a `delete_scenes(self, reference_ids: List[str])` method. We should also adapt the interface or at least add this method for the other data types to make it easier for the user and interate through the scenes internally rather then on the user side. Wdyt @gatli ?

Comment (pfmark) [reply nucleus/dataset.py:744]: Can also do this in a second PR actually. So let's go ahead and merge once the endpoint is deployed.

**Review by pfmark — APPROVED** [review]
lgtm

File: nucleus/dataset.py (hunk 1 above)
Comment (gatli) [reply nucleus/dataset.py:744]: Agreed, that is probably a more natural interface. Let's add it to shortcut!

**Pull request merged**